pkg/kds/context/context_test.go

```diff
@@ -0,0 +1,236 @@
+package context_test
+
+import (
+	stdcontext "context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	system_proto "github.com/kumahq/kuma/api/system/v1alpha1"
+	config_manager "github.com/kumahq/kuma/pkg/core/config/manager"
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	core_system "github.com/kumahq/kuma/pkg/core/resources/apis/system"
+	"github.com/kumahq/kuma/pkg/core/resources/manager"
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+	"github.com/kumahq/kuma/pkg/core/resources/registry"
+	core_store "github.com/kumahq/kuma/pkg/core/resources/store"
+	"github.com/kumahq/kuma/pkg/kds/context"
+	"github.com/kumahq/kuma/pkg/kds/reconcile"
+	"github.com/kumahq/kuma/pkg/plugins/resources/memory"
+	test_model "github.com/kumahq/kuma/pkg/test/resources/model"
+	zone_tokens "github.com/kumahq/kuma/pkg/tokens/builtin/zone"
+	"github.com/kumahq/kuma/pkg/tokens/builtin/zoneingress"
+	util_proto "github.com/kumahq/kuma/pkg/util/proto"
+)
+
+var _ = Describe("Context", func() {
+	Describe("GlobalProvidedFilter", func() {
+		var rm manager.ResourceManager
+		var predicate reconcile.ResourceFilter
+
+		clusterID := "cluster-id"
+		configs := map[string]bool{
+			config_manager.ClusterIdConfigKey: true,
+		}
+
+		BeforeEach(func() {
+			rm = manager.NewResourceManager(memory.NewStore())
+			predicate = context.GlobalProvidedFilter(rm, configs)
+		})
+
+		It("Should filter out dataplanes", func() {
+			// when
```

Comment:
nit: in `when` you should have action.

It's
given: prepare the data
when: execute the action
then: assert the result

In this case, it should be
```
// given
dp := ...

// when
filtered := predicate(clusterID, dp)

// then
Expect(filtered).To(BeFalse())
```

either
1) change `when` to `given`
2) drop `when, given, then`
3) extract action to `when` (example above)

I usually prefer 3) even if the test is longer, but I'm ok with any option

Reply:
addressed